lib/adapters/class-adapter.php

```diff
@@ -153,6 +169,83 @@ public function add_custom_date_range_aggregation( array $args = [] ): void {
 		$this->add_aggregation( new Custom_Date_Range( $this->dsl, $args ) );
 	}
 
+	/**
+	 * A callback to filter es args.
+	 * Adds phrase matching to the request args if it is enabled.
+	 *
+	 * @param array $es_args The request args to be filtered.
+	 * @return array The filtered request args.
+	 */
+	public function add_phrase_matching_to_es_args( array $es_args ): array {
+		if ( ! $this->get_enable_phrase_matching() ) {
+			return $es_args;
+		}
+
+		$search = get_query_var( 's' );
+
+		// Bail early if this isn't a search.
+		if ( empty( $search ) ) {
+			return $es_args;
+		}
+
+		// Break down the search string into the desired "phrase matched" parts.
+		// TODO: Add filter for phrase_match_delineator.
+		$phrase_match_delineator = '"';
+		preg_match_all( '/' . $phrase_match_delineator . '(.*?)' . $phrase_match_delineator . '/', $search, $matches );
+		if ( empty( $matches[1] ) ) {
+			return $es_args;
+		}
```

Comment:
imo we don't need to filter it, double quote is standard for this. Also, you can check the results of `preg_match_all` rather than looking at the first item being empty, since it will return `0` on no match or `false` on failure, and a falsey return value from that function means no matches and you can bail.
```suggestion
		$phrase_match_delineator = '"';
		if ( ! preg_match_all( '/' . $phrase_match_delineator . '(.*?)' . $phrase_match_delineator . '/', $search, $matches ) ) {
			return $es_args;
		}
```

Reply:
I will pull the filter out!